docs/csharp/tutorials/exploration/csharp-7.yml

```
@@ -0,0 +1,536 @@
+    values found in a sequence of integers:
+
+    [!code-csharp[TupleReturningMethod](~/samples/snippets/csharp/new-in-7/program.cs#08_TupleReturningMethod "Tuple returning method")]
+
+    Using tuples in this way offers several advantages:
+
+    * You save the work of authoring a `class` or a `struct` that defines the type returned. 
+    * You do not need to create new type.
+    * The language enhancements removes the need to call the <xref:System.Tuple.Create``1(``0)> methods.
+
+    The declaration for the method provides the names for the fields of the
+    tuple that is returned. When you call the method, the return value is a 
+    tuple whose fields are `Max` and `Min`:
+
+    [!code-csharp[CallingTupleMethod](~/samples/snippets/csharp/new-in-7/program.cs#09_CallingTupleMethod "Calling a tuple returning method")]
+
+    There may be times when you want to unpackage the members of a tuple that
+    were returned from a method.  You can do that by declaring separate variables
+    for each of the values in the tuple. This is called *deconstructing* the tuple:
+
+    [!code-csharp[CallingWithDeconstructor](~/samples/snippets/csharp/new-in-7/program.cs#10_CallingWithDeconstructor "Deconstructing a tuple")]
+
+    You can also provide a similar deconstruction for any type in .NET. This is
+    done by writing a `Deconstruct` method as a member of the class. That
+    `Deconstruct` method provides a set of `out` arguments for each of the
+    properties you want to extract. Consider
+    this `Point` class that provides a deconstructor method that extracts
+    the `X` and `Y` coordinates:
+
+    [!code-csharp[PointWithDeconstruction](~/samples/snippets/csharp/new-in-7/point.cs#11_PointWithDeconstruction "Point with deconstruction method")]
+    
+    You can extract the individual fields by assigning a `Point` to a tuple:
+
+    [!code-csharp[DeconstructPoint](~/samples/snippets/csharp/new-in-7/program.cs#12_DeconstructPoint "Deconstruct a point")]
+
+    You are not bound by the names defined in the `Deconstruct` method. You
+    can rename the extract variables as part of the assignment:  
+
+    [!code-csharp[DeconstructNames](~/samples/snippets/csharp/new-in-7/program.cs#13_DeconstructNames "Deconstruct with new names")]
+
+    You can learn more in depth about tuples in the
+    [tuples topic](../tuples.md).
+
+- title: Use discards to select needed members
+  durationInMinutes: 2
+  content: |
+    Often when deconstructing a tuple or calling a method with `out` parameters, you're forced to define a variable whose value you don't care about and don't intend to use. C# adds support for *discards* to handle this scenario. A discard is a write-only variable whose name is `_` (the underscore character); you can assign all of the values that you intend to discard to the single variable. A discard is like an unassigned variable; apart from the assignment statement, the discard can't be used in code.
```

```suggestion
    Often when deconstructing a tuple or calling a method with `out` parameters, you're forced to define a variable whose value you don't care about and don't intend to use. C# adds support for *discards* to handle this scenario. A discard is a write-only variable whose name is `_` (the underscore character); you can assign all of the values that you intend to discard to the single variable. A discard is like an unassigned variable; apart from the assignment, the discard can't be used in code.
```
There is no statement in a formal sense; it's a predefined variable.